packages/dds/tree/src/test/rebaserAxiomaticTests.ts

```diff
@@ -380,6 +654,196 @@ function verifyComposeAssociativity<TChangeset>(
+						editRebasedOverC,
+					),
+				),
+			]),
+		),
+	);
+
+	assertDeepEqual(actualChange, expectedChange);
+}
+
+function verifyRebaseOverUndoRedoPair<TChangeset>(
+	edit: TaggedChange<TChangeset>,
+	namedEditsToRebaseOver: NamedChangeset<TChangeset>[],
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const assertDeepEqual = getDefaultedEqualityAssert(fieldRebaser);
+	assert(namedEditsToRebaseOver.length === 1, "expected just 1 edit");
+	const editsToRebaseOver = namedEditsToRebaseOver.map(({ changeset }) => changeset);
+	const editB = { change: editsToRebaseOver[0].change, revision: editsToRebaseOver[0].revision };
+
+	const inverseEditB = fieldRebaser.invert(editB);
+
+	// ((A ↷ B) ↷ B⁻¹) ↷ B
+	const actualChange = makeAnonChange(
+		fieldRebaser.rebase(
+			fieldRebaser.rebase(
+				fieldRebaser.rebase(edit.change, editB),
+				tagChange(inverseEditB, editB.revision),
+			),
+			editB,
+		),
+	);
+
+	// A ↷ B
+	const expectedChange = makeAnonChange(fieldRebaser.rebase(edit.change, editB));
+	assertDeepEqual(actualChange, expectedChange);
+}
+
+function verifyRebaseOverDoUndoPairIsNoOp<TChangeset>(
+	edit: TaggedChange<TChangeset>,
+	namedEditsToRebaseOver: NamedChangeset<TChangeset>[],
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const assertDeepEqual = getDefaultedEqualityAssert(fieldRebaser);
+	assert(namedEditsToRebaseOver.length === 1, "expected just 1 edit");
+	const editsToRebaseOver = namedEditsToRebaseOver.map(({ changeset }) => changeset);
+
+	const editB = { change: editsToRebaseOver[0].change, revision: editsToRebaseOver[0].revision };
+
+	const invertedEditB = fieldRebaser.invert(editB);
+	// (A ↷ B) ↷ B⁻¹
+	const actualChange = makeAnonChange(
+		fieldRebaser.rebase(
+			fieldRebaser.rebase(edit.change, editB),
+			tagChange(invertedEditB, editB.revision),
+		),
+	);
+
+	const expectedChange = makeAnonChange(edit.change);
+	if (fieldRebaser.compareWithoutTombstones !== undefined) {
+		fieldRebaser.compareWithoutTombstones(actualChange.change, expectedChange.change);
+	} else {
+		assertDeepEqual(actualChange, expectedChange);
+	}
+}
+
+function verifyRebaseOverEmpty<TChangeset>(
+	edit: TaggedChange<TChangeset>,
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const emptyChange = tagChange(
+		fieldRebaser.createEmpty(),
+		"test" as unknown as SessionSpaceCompressedId,
+	);
+	const actualChange = makeAnonChange(fieldRebaser.rebase(edit.change, emptyChange));
+	const expectedChange = makeAnonChange(edit.change);
+	const assertDeepEqual = getDefaultedEqualityAssert(fieldRebaser);
+	assertDeepEqual(actualChange, expectedChange);
+}
+
+function verifyRebaseEmpty<TChangeset>(
+	edit: TaggedChange<TChangeset>,
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const emptyChange = tagChange(
+		fieldRebaser.createEmpty(),
+		"test" as unknown as SessionSpaceCompressedId,
+	);
+
+	const actualChange = makeAnonChange(fieldRebaser.rebase(emptyChange.change, edit));
```

Comment:
This should use the same tag as the input. We can maybe use `rebaseTagged` to avoid this issue.

Reply:
Updated to use `rebaseTagged`